### PR TITLE
fix(zebrad) print usage info for --help flag

### PR DESCRIPTION
**Pull request by arya2**
## Motivation

The `zebrad --help` currently runs the process and does not print usage info.

Related to #5624

## Solution

- Print usage info and exit the process if the `--help` flag is set.
- Set tracing filter to "warn" if the `--help` flag is set to avoid info-level logs from component registration
- Update help text for `help` command to mention top-level and subcommand options

## Review

Anyone can review this PR.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

- Print the same for invalid commands and options
- Print the same for the `help` command

### CODE REVIEW

**Comment by arya2** [comment]
@Mergifyio update

**Comment by arya2** [comment]
ssh error:

> Writing 3 keys to /home/runner/.ssh/google_compute_known_hosts
> sa_106835868714082647983@35.232.220.108: Permission denied (publickey).
> 
> ...
> 
> ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return code [255].
> Error: Process completed with exit code 255.

https://github.com/ZcashFoundation/zebra/actions/runs/3481807695/jobs/5823931796#step:9:103

**Comment by arya2** [comment]
@Mergifyio update

File: README.md [hunk 1]
```diff
@@ -242,6 +242,8 @@ There are a few bugs in Zebra that we're still working on fixing:
 - Experimental Tor support is disabled until [`arti-client` upgrades to `x25519-dalek` 2.0.0 or later](https://github.com/ZcashFoundation/zebra/issues/5492)
   - This happens due to a Rust dependency conflict, which can only be resolved by changing the dependencies of `x25519-dalek`
 
+- Output of `help`, `--help` flag, and usage of invalid commands or options are inconsistent.
```
Comment (oxarbitrage) [inline README.md]: If we have an older ticket we should add it here too.
```suggestion
- Output of `help`, `--help` flag, and usage of invalid commands or options are inconsistent. Reports of this issues can be found [here](https://github.com/ZcashFoundation/zebra/issues/5502) and are planned to be fixed in the context of [upgrading Abscissa](https://github.com/ZcashFoundation/zebra/issues/5502).
```

**Review by oxarbitrage — DISMISSED** [review]
Looks good, i can confirm one of the issues is actually fixed. But there are other problems where we want to update Abscissa to see if they get fixed by that alone. We know about this inconsistencies since ever but i am unsure if we have an older ticket for it.

I think is a good idea to have this in the known issues of the readme.

**Review by oxarbitrage — APPROVED** [review]
thanks.

**Comment by mpguerra** [comment]
@mergifyio update

**Pull request merged**